src/hooks/usePreloadAssets.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useRef } from "react";
+
+import { isDefined } from "utils/types";
+
+export interface PreloadAsset {
+  url: string;
+  as?: string;
+  type?: string;
```

Comment:
I guess these can be more strictly defined?

Reply:
Not really any benefit in it.

It's passed directly to the HTML element (declared `interface HTMLLinkElement extends HTMLElement, LinkStyle`) which has them as strings.

And I would consider it OOS to have one more think to keep track of such a thing as constraining something so basic it wasn't constrained even in the standard library.

Reply:
@goran-peoski-work if they are defined as strings in the HTML default types, then It's all good. Thanks for the research